matchms/importing/load_from_msp.py

```diff
@@ -76,58 +76,64 @@ def parse_msp_file(filename: str) -> Generator[dict, None, None]:
 
             if contains_metadata(rline):
                 parse_metadata(rline, params)
-            else:
-                # Obtaining the masses and intensities
-                peak_pairs = get_peak_tuples(rline)
-
-                for peak in peak_pairs:
-                    mz, intensity = get_peak_values(peak)
-                    comment = get_peak_comment(peak)
-                    if comment is not None:
-                        peak_comments.update({mz: comment})
-
-                    peakscount += 1
-                    masses.append(mz)
-                    intensities.append(intensity)
-
-                # Obtaining the masses and intensities
-                if int(params['num peaks']) == peakscount:
-                    peakscount = 0
-                    yield {
-                        'params': (params),
-                        'm/z array': np.array(masses),
-                        'intensity array': np.array(intensities),
-                        'peak comments': peak_comments
-                    }
-
-                    params = {}
-                    masses = []
-                    intensities = []
-                    peak_comments = {}
+                continue
 
+            mz, ints, comment = _parse_line(rline)
 
-def get_peak_values(peak: str) -> Tuple[float, float]:
-    """ Get the m/z and intensity value from the line containing the peak information. """
-    splitted_line = peak.split(maxsplit=2)
-    mz = float(splitted_line[0].strip())
-    intensity = float(splitted_line[1].strip())
-    return mz, intensity
+            masses = np.append(masses, mz)
+            intensities = np.append(intensities, ints)
+            
+            if comment is not None:
+                peak_comments.update({masses[-1]: comment})
+
+            peakscount += len(mz)
+
+
+            # Obtaining the masses and intensities
+            if int(params['num peaks']) == peakscount:
+                peakscount = 0
+                yield {
+                    'params': (params),
+                    'm/z array': masses,
+                    'intensity array': intensities,
+                    'peak comments': peak_comments
+                }
 
+                params = {}
+                masses = []
+                intensities = []
+                peak_comments = {}
+            
 
-def get_peak_tuples(rline: str) -> Iterator[str]:
-    """ Splits line at ';' and performs additional string cleaning. """
-    tokens = filter(None, rline.split(";"))
-    peak_pairs = map(lambda x: x.lstrip().rstrip(), tokens)
-    return peak_pairs
 
 
-def get_peak_comment(rline: str) -> str:
+def _parse_line(rline: str):
```

Comment:
Maybe rename to _parse_line_with_peaks. (this line should always contain mz and intensities right?

Reply:
True - will address that. A line without any mz and intensities should throw an error I guess, also to avoid silently ignoring peaks or not reading spectra properly.

Reply:
Adressed in https://github.com/matchms/matchms/pull/393/commits/7ef2c57896b08fa74c8128d038f65eed9657c820